### DISCUSSION TITLE
debug: continuous output on serial while mg_log_set(MG_LL_NONE);

### DISCUSSION THREAD

**Discussion by Mike0675** [cesanta/mongoose#2438] [Ask questions, get answers] [answered]
Hi,
I´m tryin to get Teensy 4.1 get to work with w5500 driver.

http-server works, but mqtt don´t.

While looking what mongooses debug says about that, I can´t see anything couse serial output is filled with continuous HEX.

I set mg_log_set to MG_LL_NONE so there should be no output at all.

My setting is:

`       mg_log_set_fn([](char ch, void *)
				  { Serial8.print(ch); },
				  NULL);
	mg_log_set(MG_LL_NONE);

	// Init Manager
	mg_mgr_init(&mgr);
	delay(3000);
	MG_INFO(("Starting TCP/IP stack..."));

	mif.driver = &mg_tcpip_driver_w5500;
	mif.driver_data = &spi;
	mg_tcpip_init(&mgr, &mif);

	mif.enable_dhcp_client = false;
	mif.enable_dhcp_server = false;

	// mif.mac = mac;
	mif.ip = ip;
	mif.mask = subnet;
	mif.gw = gateway;`

**Comment by scaprile** [comment, marked as answer]
If there is hex in your log, then find where mg_hexdump is called and comment it out
Please follow the guidelines in the [documentation](https://mongoose.ws/documentation), [examples](https://github.com/cesanta/mongoose/tree/master/examples) and [tutorials](https://mongoose.ws/tutorials/) available.
